subprojects/configuration-cache-report/src/main/kotlin/ConfigurationCacheReportPage.kt

```diff
@@ -186,45 +165,74 @@ object ConfigurationCacheReportPage : Component<ConfigurationCacheReportPage.Mod
     fun viewProblems(model: Model) = div(
         attributes { className("content") },
         when (model.tab) {
-            Tab.ByMessage -> viewTree(model.messageTree, Intent::MessageTreeIntent, model.displayFilter)
-            Tab.ByLocation -> viewTree(model.locationTree, Intent::TaskTreeIntent, model.displayFilter)
+            Tab.Inputs -> viewInputs(model.inputTree)
+            Tab.ByMessage -> viewTree(model.messageTree, Intent::MessageTreeIntent)
+            Tab.ByLocation -> viewTree(model.locationTree, Intent::TaskTreeIntent)
         }
     )
 
     private
-    fun viewInputs(inputTree: ProblemTreeModel): View<Intent> = when (inputTree.problemCount) {
-        0 -> h1("No build logic inputs were detected.")
-        else -> div(
-            div(
-                attributes { className("title") },
-                h1("The following build logic inputs were automatically detected and will cause the cache to be discarded when their value change:"),
-            ),
-            div(
-                attributes { className("inputs") },
-                viewTree(inputTree.tree.focus().children, Intent::InputTreeIntent)
-            )
+    fun viewInputs(inputTree: ProblemTreeModel): View<Intent> =
+        div(
+            attributes { className("inputs") },
+            viewTree(inputTree.tree.focus().children, Intent::InputTreeIntent)
+        )
+
+    private
+    fun displaySummary(model: Model): View<Intent> {
+        return h1(
+            "${model.cacheAction.capitalize()} the configuration cache for ",
+            code(model.requestedTasks),
+            br(),
+            small(model.inputsSummary()),
+            br(),
+            small(model.problemsSummary()),
         )
     }
 
     private
-    fun Model.summary() =
-        "$totalProblems ${problemOrProblems()} found $cacheAction the configuration cache".let {
-            if (totalProblems > reportedProblems) "$it, only the first $reportedProblems were included in this report"
+    fun Model.inputsSummary() =
+        found(inputTree.problemCount, "build configuration input").let {
+            if (inputTree.problemCount > 0) "$it and will cause the cache to be discarded when ${itsOrTheir(inputTree.problemCount)} value change"
+            else it
+        }
+
+    private
+    fun Model.problemsSummary() =
+        found(totalProblems, "problem").let {
+            if (totalProblems > reportedProblems) "$it, only the first $reportedProblems ${wasOrWere(reportedProblems)} included in this report"
             else it
         }
 
     private
-    fun Model.problemOrProblems() =
-        if (totalProblems == 1) "problem was" else "problems were"
+    fun found(count: Int, what: String) =
+        "${count.toStringOrNo()} ${what.sIfPlural(count)} ${wasOrWere(count)} found"
+
+    private
+    fun Int.toStringOrNo() =
+        if (this != 0) toString()
```

Comment:
minor 💄 - the positive test would read slightly better `if (this == 0) "No" else toString()`

Reply:
I wrote it with a positive test at first but changed it so it reads the same as the function name.

Reply:
What about?
```kotlin
fun Int.toStringOrNo() =
     if (this > 0) toString()
     else "No"
```

Reply:
Then I'd fear negative values could be missed if we ever introduce a related bug :D

Not big deal, it's good enough as it is 👍